core/src/main/java/com/novoda/landingstrip/LandingStripAdapter.java

```diff
@@ -0,0 +1,38 @@
+package com.novoda.landingstrip;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+public abstract class LandingStripAdapter<T extends View> {
+
+    private Listener listener;
+
+    protected abstract T createView(ViewGroup parent, int position);
+
+    protected abstract void bindView(T view, int position);
+
+    void setListener(Listener listener) {
+        this.listener = listener;
+    }
+
+    public void notifyDataSetChanged() {
+        if (listener != null) {
+            listener.onNotifyDataSetChanged(this);
+        }
+    }
+
+    public void notifyItemChanged(int position) {
+        if (listener != null) {
+            listener.onNotifyItemChanged(this, position);
+        }
+    }
+
+    protected abstract int getCount();
+
+    interface Listener {
+
+        <T extends View> void onNotifyDataSetChanged(LandingStripAdapter<T> adapter);
```

Comment:
>Perhaps it would be better to create the `listener<T>` in `setAdapter` instead of allowing dynamic types

Yeah I'd rather do that, seems more correct given that is the scope where you know the types